src/wp-includes/class-wp-user-query.php

```diff
@@ -804,28 +804,41 @@ public function query() {
 				{$this->query_orderby}
 				{$this->query_limit}
 			";
-
-			if ( is_array( $qv['fields'] ) ) {
-				$this->results = $wpdb->get_results( $this->request );
+			$cache_key   = $this->generate_cache_key( $qv, $this->request );
+			$cache_value = wp_cache_get( $cache_key, 'users' );
```

Comment:
`$cache_key` seems like it'll be unique here. I'm noting this because the `'users'` group is also used for storing WP_User data. In that case, it's keyed by `$user_id` so can't conflict.